### PR TITLE
[sw/boot_rom] Rename "boot_rom" to "test_rom" to fix #9154.

### LINKED ISSUE #9154: [sw/boot_rom] Rename "boot_rom" to "test_rom"
The ROM code in `sw/device/boot_rom/` is used purely for testing purposes, and is not designed to be used as the OT mask ROM. The mask ROM code lives in `sw/device/silicon_creator/mask_rom`. To make this more clear, I think it would help if `sw/device/boot_rom` was renamed and relocated to `sw/device/lib/testing/test_rom`, or something similar.

Thoughts?

### PR DESCRIPTION
**Pull request by timothytrippel**
The ROM code in `sw/device/boot_rom/` is used purely for testing purposes, and is not designed to be used as the production mask ROM. The mask ROM code lives in `sw/device/silicon_creator/mask_rom`. To make this more clear, the `sw/device/boot_rom` ia renamed and relocated to `sw/device/lib/testing/test_rom`.

This fixes #9154 (which was discussed in the SW WG).

Signed-off-by: Timothy Trippel <ttrippel@google.com>

### CODE REVIEW

**Comment by timothytrippel** [comment]
CC @domrizz0 (you're wish has finally come true!)

**Review by msfschaffner — APPROVED** [review]
Thanks for the cleanup!
Probably good to get an LGTM from a SW or DV person as well.

**Review by mcy — APPROVED** [review]

**Comment by timothytrippel** [comment]
Please wait to merge, I want to rebase this on top of #9412 (once it merges).

File: hw/top_englishbreakfast/util/sw_sources.patch
Comment (alphan) [inline hw/top_englishbreakfast/util/sw_sources.patch:73]: Can there be unintended changes in this file?

**Review by alphan — APPROVED** [review]
Took a quick look and LGTM except for a couple of changes in `sw_sources.patch` that may be unintentional. Would you mind taking another look @timothytrippel?

File: hw/top_englishbreakfast/util/sw_sources.patch
Comment (timothytrippel) [reply hw/top_englishbreakfast/util/sw_sources.patch]: Thanks @alphan, good catch! Fixed.

**Review by cfrantz — APPROVED** [review]

**Comment by timothytrippel** [comment]
This PR is now ready to merge.

**Pull request merged**
